test/functional/test_framework/key.py

```diff
@@ -27,6 +27,22 @@ def TaggedHash(tag, data):
     ss += data
     return hashlib.sha256(ss).digest()
 
+# HKDF-SHA256
+def hmac_sha256(key, data):
```

Comment:
As `key.py` is currently containing only code related to digital signatures (according to the module description it's a "secp256k1 elliptic curve protocols implementation"), the HKDF would maybe be better suited in another (new?) module. Probably it generally makes sense to do a follow-up with restructuring crypto-related helpers, e.g. having an own "test_framework/crypto" folder would be nice on the long-term for better structuring I think, as the list of crypto-related modules has grown quite a bit.

Reply:
done. used a separate `crypto/` module.